src/main/java/DianaAssistant.java

```diff
@@ -0,0 +1,155 @@
+        try {
+            int TaskNum = Integer.valueOf(input.substring(input.indexOf(" ") + 1)) - 1;
+            if (TaskNum >= 0 && TaskNum < list.size()) {
+                Task task = list.get(TaskNum);
+                if (toMark) {
+                    task.markAsDone();
+                    printEnclosure();
+                    System.out.println("Nice! I've marked this task as done\n" + task.toString());
+                    printEnclosure();
+                }
+                else {
+                    task.markAsNotDone();
+                    printEnclosure();
+                    System.out.println("Okay, I've marked this task as not done yet\n" + task.toString());
+                    printEnclosure();
+                }
+            }
+            else {
+                System.out.println("Please enter a number between 1 and " + (list.size() - 1));
+            }
+        }
+        catch (NumberFormatException e) {
+            System.out.println("Please enter a number between 1 and " + (list.size() - 1));
+        }
+    }
+
+    private void printAddedTask (String description, List<Task> list) {
+        printEnclosure();
+        System.out.println("Got it! I've added this task");
+        System.out.println(description);
+        System.out.println("Now you have " + list.size() + " tasks in the list");
+        printList(list);
+    }
+
+    private void addTodo (String input, List<Task> list) {
+        String description = input.substring("todo".length()).trim();
+        Todo todo = new Todo(description);
+        list.add(todo);
+        printAddedTask(description, list);
+    }
+
+    private void addDeadline (String input, List<Task> list) {
+        if (!input.contains("/by")) {
+            System.out.println("Please enter a valid deadline with the correct format:");
+            System.out.println("Eg: Read book /by 5pm");
+            return;
+        }
+        String[] parts = input.split("/by", 2);
+        String description = parts[0].substring("deadline".length()).trim();
+        String by = parts[1].trim(); // trim removes white space
+        Task deadline = new Deadline(description, by);
+        list.add(deadline);
+        printAddedTask(description, list);
+    }
+
+    private void addEvent (String input, List<Task> list) {
+        if (!(input.contains("/from") && input.contains("/to"))) {
+            System.out.println("Please enter a valid event with the correct format:");
+            System.out.println("Eg: Read book /from 2pm /to 4pm");
+            return;
+        }
+
+        String[] parts = input.split("/from", 2);
+        String description = parts[0].substring("event".length()).trim();
+        String[] fromto = parts[1].split("/to", 2);
+        String from = fromto[0].trim();
+        String to = fromto[1].trim();
+        Task event = new Event(description, from, to);
+        list.add(event);
+        printAddedTask(description, list);
```

Comment:
Could you put this block of code within an else block, to make it more obvious that it is executed only when the input string contains "/from" and "/to"?